app/bin/tools/remove_package.dart

```diff
@@ -69,13 +70,25 @@ Future listPackage(String packageName) async {
   }
 }
 
-Future removePackage(String packageName) async {
+Future _deleteWithQuery(Query query) async {
   final deletes = <Key>[];
-  final jobQuery = dbService.query(Job)..filter('packageName =', packageName);
-  await for (Job job in jobQuery.run()) {
-    deletes.add(job.key);
+  await for (Model m in query.run()) {
+    deletes.add(m.key);
   }
+  await dbService.withTransaction((tx) async {
+    tx.queueMutations(deletes: deletes);
+    await tx.commit();
+  });
+}
+
+Future removePackage(String packageName) async {
+  await _deleteWithQuery(
+      dbService.query(Job)..filter('packageName =', packageName));
```

Comment:
Should we do this after deleting the Package? There's a tiny window here where our scheduling job might pick up the package and add new Jobs for it, after we've deleted the Jobs, but before we delete the Package/PackageVersion.

Reply:
Moved both to the end of the method. I believe there is always a window that could introduce artifacts, e.g. when there is an ongoing analysis and we delete the records while it is still doing its stuff. The script can be run multiple times though, cleaning these up later.